packages/components/src/components/common/table/hooks/hooks.tsx

```diff
@@ -209,10 +212,10 @@ export const useTableKeyHandler = <TData,>({ table, data, options }: TableKeyboa
 };
 
 const toggleExpand = <TData,>(row: Row<TData>, expand: boolean, loadChildren?: (row: Row<TData>) => void) => {
-  if (expand && !row.getIsExpanded()) {
+  if (expand && row.getIsExpanded && !row.getIsExpanded()) {
```

Comment:
why needed?

Reply:
I had some strange getIsExpanded is undefined errors in the browser. But I couldn't reproduce them today. However, I just got an "row is undefined" error when trying to use the arrowleft/right button while no row is selected. I will now also check if the row is not undefined and can be expanded.

Reply:
the point is that typescript should make sure that the row is not undefined.
So either we should mark the row here as optional, so typescript can help us or identify the real problem and check it there.
To be clear, I think this is the array get syntax which is not clear that it can also return undefined. So I think we should mark the row here as undefined and check it.

Reply:
Not 100% sure if i understood correctly... I marked the row as optional now and check at the start, if the row is undefined. Something like this:
![grafik](https://github.com/user-attachments/assets/4ca3faf6-63fb-4cdf-ae9a-bcdbc4f8d710)

Reply:
When we don't mark the row as optional here then what happens? We lie to ourselfes and don't help anyone which uses this code.

Typescript is only a layer on top of JS, for js it don't makes any difference if this variable is define or not, but in the editor we want to have the best experience. and we don't get there when we lie to ourselves. This row can be undefined (even if TypeScript is not clever enough to recognize that), so we should help ourself by mark it like that so then typescript can help us catch errors.